x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/watch/WatchTests.java

```diff
@@ -296,6 +303,31 @@ public void testParserBadActions() throws Exception {
         }
     }
 
+    public void testParserConsumesEntireDefinition() throws Exception {
+        WatchParser wp = createWatchparser();
+        XContentBuilder jsonBuilder = jsonBuilder()
+            .startObject()
```

Comment:
Any reason you did not opt for the syntax (yea i know, its a builder, lets chain it!!!) used in other spots in this class? Imho its harder to read the nested thing than blocks of code with spaces that define start/endings. This is also how we do it in most of our toXContent functions as well.

```
b = jsonBuilder();
b.startObject();

b.startObject("trigger");
b.startObject("schedule");
b.field("interval", "10s");
b.endObject();
b.endObject();

...

b.endObject();
```

Reply:
We prefer using braces like this:

```diff
diff --git a/x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/watch/WatchTests.java b/x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/watch/WatchTests.java
index 1ff59034831..dcb48fbd3c1 100644
--- a/x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/watch/WatchTests.java
+++ b/x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/watch/WatchTests.java
@@ -305,27 +305,44 @@ public class WatchTests extends ESTestCase {
 
     public void testParserConsumesEntireDefinition() throws Exception {
         WatchParser wp = createWatchparser();
-        XContentBuilder jsonBuilder = jsonBuilder()
-            .startObject()
-              .startObject("trigger")
-                .startObject("schedule")
-                  .field("interval", "10s")
-                .endObject()
-              .endObject()
-              .startObject("input")
-                .startObject("simple")
-                .endObject()
-              .endObject()
-              .startObject("condition")
-                .startObject("script")
-                  .field("source", "return false")
-                .endObject()
-              .endObject()
-            .endObject();
-            jsonBuilder.generator().writeBinary(",".getBytes(StandardCharsets.UTF_8));
-            ElasticsearchParseException e = expectThrows(ElasticsearchParseException.class, () ->
-                wp.parseWithSecrets("failure", false, BytesReference.bytes(jsonBuilder), new DateTime(), XContentType.JSON, true));
+        try (XContentBuilder builder = jsonBuilder()) {
+            builder.startObject();
+            {
+                builder.startObject("trigger");
+                {
+                    builder.startObject("schedule");
+                    {
+                        builder.field("interval", "10s");
+                    }
+                    builder.endObject();
+                }
+                builder.endObject();
+                builder.startObject("input");
+                {
+                    builder.startObject("simple");
+                    {
+                    }
+                    builder.endObject();
+                }
+                builder.endObject();
+                builder.startObject("condition");
+                {
+                    builder.startObject("script");
+                    {
+                        builder.field("source", "return false");
+                    }
+                    builder.endObject();
+                }
+                builder.endObject();
+            }
+            builder.endObject();
+            builder.generator().writeBinary(",".getBytes(StandardCharsets.UTF_8));
+            ElasticsearchParseException e = expectThrows(
+                    ElasticsearchParseException.class,
+                    () -> wp.parseWithSecrets("failure", false, BytesReference.bytes(builder), new DateTime(), XContentType.JSON, true));
             assertThat(e.getMessage(), containsString("unexpected data beyond"));
+        }
+
     }
 
     public void testParserDefaults() throws Exception {
```

Also note that the builder should be wrapped in a try-with-resources; it's good hygiene.

Reply:
Oh, ive not see this before but i like it. I almost removed a `builder.endObject()` on one of my first XContent forays, and I dont think i would have even considered it if this syntax was used.

Also, WRT the try-with-resources and hygiene, this is specifically for tests? or should we be doing this in the XContent methods as well? I assumed the caller was responsible for it in our request/response objects.

Reply:
Here we own the lifecycle of the builder so we are responsible for closing it.

Reply:
@hub-cap The way the class does it is mixed, when i started this PR i based style on testParserBadActions(), as that seemed to be the closest example to what I wanted to do.  I am happy to conform to whatever is preferred.

Reply:
That's the old style before we introduced using braces for this.